modules/AlphaWalletFoundation/AlphaWalletFoundation/RPC/BlockchainProvider.swift

```diff
@@ -21,11 +23,12 @@ public protocol BlockchainProvider {
     func call(from: AlphaWallet.Address?, to: AlphaWallet.Address?, value: String?, data: String) -> AnyPublisher<String, SessionTaskError>
     func call<R: ContractMethodCall>(_ method: R, block: BlockParameter) -> AnyPublisher<R.Response, SessionTaskError>
     func pendingTransaction(hash: String) -> AnyPublisher<EthereumTransaction?, SessionTaskError>
-    func nextNonce(wallet: AlphaWallet.Address) -> AnyPublisher<Int, SessionTaskError>
+    func nonce(wallet: AlphaWallet.Address) async throws -> Int
```

Comment:
`getNextNonce(wallet:)` seems better? "next" is more intention-revealing, and "get" shows it's doing something.